### PR TITLE
MinGW SDLmain fix

### PR DESCRIPTION
**Pull request by arrowgent**
https://github.com/diasurgical/devilutionX/issues/686

compiling out of order SDLmain

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ cmake --build . -j $(sysctl -n hw.ncpuonline)
 ### Installing dependencies on WSL, Debian and Ubuntu
 
 Download and place the 32bit MinGW Development Libraries of [SDL2](https://www.libsdl.org/download-2.0.php), [SDL2_mixer](https://www.libsdl.org/projects/SDL_mixer/), [SDL2_ttf](https://www.libsdl.org/projects/SDL_ttf/) and [Libsodium](https://github.com/jedisct1/libsodium/releases) in `/usr/i686-w64-mingw32`.
+NOTE: SDL2-devel-2.0.9-mingw.tar.gz required
```
Comment (AJenbo) [inline README.md:86]: it should compile fine all the way back to SDL2 2.0.4

File: CMakeLists.txt [hunk 2]
```diff
@@ -447,9 +447,9 @@ foreach(target devilution ${BIN_TARGET})
     target_compile_definitions(${target} PRIVATE USE_SDL1)
   else()
     target_link_libraries(${target} PRIVATE
+      SDL2::SDL2main
```
Comment (AJenbo) [inline CMakeLists.txt:450]: Curious that you need SDL2main specified first. I just build it on a clean Ubuntu 18.04 and didn't get any errors. Also Amiga requires it to be last (uses SDL1) because it's statically linking.

File: README.md (hunk 1 above)
Comment (arrowgent) [reply README.md:86]: doesnt compile with SDL2-devel-2.0.12-mingw.tar.gz
it fails at 64%

File: CMakeLists.txt (hunk 2 above)
Comment (arrowgent) [reply CMakeLists.txt:450]: http://www.cplusplus.com/forum/beginner/110753/

researched this and did

grep -r SDLmain to find it was located in CMakeLists.txt

this fixed the compile error
https://github.com/diasurgical/devilutionX/issues/686#issuecomment-609111543

**Comment by arrowgent** [comment]
please note i was compiling with release v1.0.1 not the latest master branch

File: README.md (hunk 1 above)
Comment (AJenbo) [reply README.md:86]: Yeah, I hear that 2.0.12 fails to compile on a lot of platforms, macOS also fails by default. But the text you added is misleading, especially since the issue is with SDL and not us.

Comment (AJenbo) [reply README.md:86]: Maybe rephrase it as 2.0.4-2.0.10 tested/supported, or specifically mention that 2.0.12 has issues.

File: CMakeLists.txt (hunk 2 above)
Comment (arrowgent) [reply CMakeLists.txt:450]: > Curious that you need SDL2main specified first. I just build it on a clean Ubuntu 18.04 and didn't get any errors. Also Amiga requires it to be last (uses SDL1) because it's statically linking.
seems to have built amiga fine
https://github.com/diasurgical/devilutionX/pull/687/checks?check_run_id=561411594

> amigaos-m68k - Success

File: README.md (hunk 1 above)
Comment (arrowgent) [reply README.md:86]: trying to find a way to clarify this since the readme currently only specifies to go download these libraries, by default the latest versions

File: CMakeLists.txt (hunk 2 above)
Comment (AJenbo) [reply CMakeLists.txt:450]: Yeah as mentioned Amiga uses SDL1 and you didn't change the order of it. But I'm wondering what would happen if SDL2 was statically linked in the same way on a a plaform.

File: README.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ cmake --build . -j $(sysctl -n hw.ncpuonline)
 
 ### Installing dependencies on WSL, Debian and Ubuntu
 
-Download and place the 32bit MinGW Development Libraries of [SDL2](https://www.libsdl.org/download-2.0.php), [SDL2_mixer](https://www.libsdl.org/projects/SDL_mixer/), [SDL2_ttf](https://www.libsdl.org/projects/SDL_ttf/) and [Libsodium](https://github.com/jedisct1/libsodium/releases) in `/usr/i686-w64-mingw32`.
+Download and place the 32bit MinGW Development Libraries of [SDL2](https://www.libsdl.org/download-2.0.php), [SDL2_mixer](https://www.libsdl.org/projects/SDL_mixer/), [SDL2_ttf](https://www.libsdl.org/projects/SDL_ttf/) and [Libsodium](https://github.com/jedisct1/libsodium/releases) in `/usr/i686-w64-mingw32`. NOTE: SDL2-devel 2.0.4-2.0.10 tested/supported
```
Comment (AJenbo) [inline README.md:85]:
```suggestion
Download and place the 32bit MinGW Development Libraries of [SDL2](https://www.libsdl.org/download-2.0.php), [SDL2_mixer](https://www.libsdl.org/projects/SDL_mixer/), [SDL2_ttf](https://www.libsdl.org/projects/SDL_ttf/) and [Libsodium](https://github.com/jedisct1/libsodium/releases) in `/usr/i686-w64-mingw32`.
NOTE: SDL2 2.0.12 appears to not compile correctly.
```
